filament/include/filament/Engine.h

```diff
@@ -725,6 +725,20 @@ class UTILS_PUBLIC Engine {
       */
     utils::JobSystem& getJobSystem() noexcept;
 
+    /**
+      * Tells the driver to reset any internal state tracking if necessary.
+      * 
+      * This should only be used in WebGL, and only if you are integrating an external
+      * renderer with Filament. This can be useful because Filament keeps track of the GL
+      * state it has set (like which texture is bound), and does not re-set that state if
+      * it does not think it needs to. However, if an external renderer has set different
+      * state in the mean time, Filament will use that new state unknowingly.
+      * 
+      * If you are in this situation, call this function - ideally only once per frame, 
+      * immediately after the external renderer has finished.
+      */
+    void resetInternalStateTracking() noexcept;
```

Comment:
Maybe use a more specific name like resetDriverState() or resetBackendState()?

Reply:
If we restrict this to only existing in `WebGL`, I could go even more specific like `resetWebGLDriverState()`

Reply:
I think `resetBackendState()` sounds good. I would just `#if defined(__EMSCRIPTEN__)` it so it doesn't even appear on regular builds.